### ISSUE TITLE
Several Undefined Behaviours and Stack Use After Scope reports when running tests

### ISSUE THREAD

**Issue by kunitoki** [vinniefalco/LuaBridge#296]
Running the tests with ASAN and UBSAN triggers tons of reports.

See https://pastebin.com/VDygjbSv (was too big to paste here).

**Comment by dmitry-t** [comment]
Interesting.

It's MacOS, so a 64 bit build.
I don't quite understand what is the issue with the 4 byte boundary.
Maybe this [reason](https://stackoverflow.com/questions/612443/why-does-the-mac-abi-require-16-byte-stack-alignment-for-x86-32)?

Also it's Lua which allocates memory for userdata, so actually LuaBridge has no control over the initial alignment.

Is it possible that Lua for tests is built incorrectly?
There's a chance a correctly built Lua will not produce such alignment issues.

**Comment by kunitoki** [comment]
Yes the problem is a misalignment of the memory returned by lua that is used as is to do placement new luabridge classes, no matter which lua version you chose.

**Comment by dmitry-t** [comment]
@kunitoki Are the tests failed on MacOS without the ASAN?

**Comment by kunitoki** [comment]
The tests are passing but if i will enable ASAN and UBSAN in an application using luabridge, it's annoying as it creates lot of noise when one should be interested in reports coming from the application itself.
